pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -231,9 +301,107 @@ func (r *ReconcileClusterPool) Reconcile(request reconcile.Request) (reconcile.R
+		log.WithError(err).Error("error reconciling RBAC")
+		return reconcile.Result{}, err
+	}
+
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileClusterPool) reconcileRBAC(
+	clp *hivev1.ClusterPool,
+	logger log.FieldLogger,
+) error {
+	rbList := &rbacv1.RoleBindingList{}
+	if err := r.Client.List(context.Background(), rbList, client.InNamespace(clp.GetNamespace())); err != nil {
+		log.WithError(err).Error("error listing rolebindings")
+		return err
+	}
+	var subs []rbacv1.Subject
+	var roleRef rbacv1.RoleRef
+	for _, rb := range rbList.Items {
+		if rb.RoleRef.Kind != "ClusterRole" ||
+			rb.RoleRef.Name != clusterPoolAdminRoleName {
+			continue
+		}
+		roleRef = rb.RoleRef
+		subs = append(subs, rb.Subjects...)
+	}
+
+	if len(subs) == 0 {
+		// nothing to do here.
+		return nil
+	}
+
+	// get namespaces where role needs to be bound.
+	cdNSList := &corev1.NamespaceList{}
+	if err := r.Client.List(context.Background(), cdNSList,
+		client.MatchingLabels{constants.ClusterPoolNameLabel: clp.GetName()}); err != nil {
+		log.WithError(err).Error("error listing namespaces for cluster pool")
+		return err
+	}
+
+	// create role bindings.
+	var errs []error
+	for _, ns := range cdNSList.Items {
+		if ns.DeletionTimestamp != nil {
+			logger.WithField("namespace", ns.GetName()).
+				Debug("skipping syncing the rolebindings as the namespace is marked for deletion")
+			continue
+		}
+		rb := &rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: ns.GetName(),
+				Name:      clusterPoolAdminRoleBindingName,
+			},
+			Subjects: subs,
+			RoleRef:  roleRef,
+		}
+		orb := &rbacv1.RoleBinding{}
+		if err := r.applyRoleBinding(rb, orb, logger); err != nil {
+			errs = append(errs, errors.Wrapf(err, "failed to apply rolebinding in namespace %s", ns.GetNamespace()))
+			continue
+		}
+	}
+	if err := utilerrors.NewAggregate(errs); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *ReconcileClusterPool) applyRoleBinding(desired, observed *rbacv1.RoleBinding, logger log.FieldLogger) error {
+	key := client.ObjectKey{Namespace: desired.GetNamespace(), Name: desired.GetName()}
+	logger = logger.WithField("rolebinding", key)
+
+	switch err := r.Client.Get(context.Background(), key, observed); {
+	case apierrors.IsNotFound(err):
+		logger.Info("creating rolebinding")
+		if err := r.Create(context.Background(), desired); err != nil {
+			logger.WithError(err).Error("could not create rolebinding")
+			return err
+		}
+		return nil
+	case err != nil:
+		logger.WithError(err).Error("error getting rolebinding")
+		return errors.Wrap(err, "could not get rolebinding")
+	}
+
+	if reflect.DeepEqual(observed.Subjects, desired.Subjects) && reflect.DeepEqual(observed.RoleRef, desired.RoleRef) {
+		return nil
+	}
+	observed.Subjects = desired.Subjects
+	observed.RoleRef = desired.RoleRef
+
+	logger.Info("updating rolebinding")
+	if err := r.Update(context.Background(), observed); err != nil {
```

Comment:
Would it be better here to use kubectl apply logic for this? If so see pkg/resource/helper.go and associated apply.go.

Reply:
I don't want to use the kubectl's apply logic here, but just the UPDATE to enforce the entire object. i think kubectl apply would be overkill here.